### ISSUE TITLE
Error copying SQL database within Elastic Pool

### ISSUE THREAD

**Issue by hovsepm** [Azure/azure-libraries-for-net#41]
_From @martijnkamphuis on September 12, 2017 11:24_

First, let me sketch my situation (resource type - resource name). All resources are in one resource group:
- SQL server - sql_server_A
--- Elastic Pool - elastic_pool_1
----- database - template_db

- SQL server - sql_server_B (no elastic pools or existing databases)

- SQL server - sql_server_C
--- Elastic Pool - elastic_pool_2

- SQL server - sql_server_D
--- Elastic Pool - elastic_pool_1 (name equal to Elastic Pool from sql_server_A)

Code being used:
```csharp
ISqlDatabase template_db = /* Template database obtained from source server */;
ISqlServer sqlTargetServer = /* SQL server instance obtained */;
var dbDefinition = sqlTargetServer.Databases.Define("db_from_template");
var db = await dbDefinition.WithSourceDatabase(template_db)
			.WithMode(CreateMode.Copy)
			.CreateAsync();
```

I am trying to copy my 'template_db' from 'sql_server_A' to any of the other SQL servers. What works is copying either to the same SQL server (sql_server_A), or to a different SQL server containing an Elastic Pool with the name equal to the one where the template database is linked to (sql_server_D, elastic_pool_1).

In the other cases (copy to sql_server_B or sql_server_C), an exception is thrown:
```
{"Long running operation failed with status 'Failed'."}
   at Microsoft.Rest.Azure.AzureClientExtensions.<GetLongRunningOperationResultAsync>d__1`2.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Rest.Azure.AzureClientExtensions.<GetLongRunningOperationResultAsync>d__0`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Rest.Azure.AzureClientExtensions.<GetPutOrPatchOperationResultAsync>d__4`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.Sql.Fluent.DatabasesOperations.<CreateOrUpdateWithHttpMessagesAsync>d__13.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.Sql.Fluent.DatabasesOperationsExtensions.<CreateOrUpdateAsync>d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.Sql.Fluent.SqlDatabaseImpl.<CreateChildResourceAsync>d__87.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.ResourceManager.Fluent.Core.IndependentChildImpl`7.<CreateResourceAsync>d__16.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.ResourceManager.Fluent.Core.ResourceActions.Creatable`4.<Microsoft-Azure-Management-ResourceManager-Fluent-Core-ResourceActions-IResourceCreator<IResourceT>-CreateResourceAsync>d__15.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.ResourceManager.Fluent.Core.DAG.CreatorTaskItem`1.<ExecuteAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.ResourceManager.Fluent.Core.DAG.TaskGroupBase`1.<ExecuteNodeTaskAsync>d__14.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Test.AzureService.<CopyDatabase>d__9.MoveNext() in C:\xxx\Test\AzureService.cs:line 132
```

In the Azure portal, it is possible to copy the database to a target server **_AND_** configure the Elastic Pool or DatabaseEdition/pricing tier. These options are not available in the Fluent SDK when using .WithSourceDatabase and/or .WithMode. I believe this should be added.

_Copied from original issue: Azure/azure-sdk-for-net#3684_

**Comment by hovsepm** [comment]
_From @martijnkamphuis on September 13, 2017 8:36_

For completeness; I'm using SDK version 1.2.0

**Comment by johnverbiest** [comment]
@hovsepm Any news regarding this issue? I'm facing the same issue as well...

**Comment by milismsft** [comment]
@johnverbiest What SDK version are you using?

**Comment by hovsepm** [comment]
@johnverbiest we have just released v1.8. Please update and let me know if the issue is fixed.

**Comment by johnverbiest** [comment]
Hi,

Using v1.8, I still get this error:

![afbeelding](https://user-images.githubusercontent.com/22546421/38089146-b6e517da-335e-11e8-89a4-f33a31ca4cd3.png)

**Comment by panesofglass** [comment]
I think this is similar to the issue I just posted, #244, though I'm also trying to go across resource groups and from an elastic pool to just a service objective.

**Comment by hovsepm** [comment]
closed as dupe of #244
